### ISSUE TITLE
rename `ast` to `messages`?

### ISSUE THREAD

**Issue by samuelstroschein** [inlang/inlang#1107]
## Problem 

`AST` and `query` are not two distinct modules after the refactoring. 

## Proposal 

Consolidate `AST` and `query` under `@inlang/core/messages`. 

- exports the `Message` type
- exports the `query` implementation for `setupConfig` that returns the `messages` object

**Comment by NilsJacobsen** [comment]
Does the `get` function also live there?

example:
`const message = messages.get({ id: "hello.login", languageTag: "en-US" })`

**Comment by samuelstroschein** [comment]
hey rollback, i think a separate `query` function is good: 

- whatever we come up with in #1112, we will lock ourselves into it. a separate query function can be replaced in the future with `query2` after we have better requirements
- bundling query with `messages` will lead to complexity (and lock-in) in setupConfig and locks setup config to the query schema in stead of raw data type

**Comment by felixhaeberle** [comment]
> bundling query with messages will lead to complexity (and lock-in) in setupConfig and locks setup config to the query schema in stead of raw data type

On a second note, yes. That's true. *Maybe this was a too fast by-producrt by our previous discussion.*

**Comment by samuelstroschein** [comment]
An idea that just came to my mind: 

### Proposal - Return messages as object similar to #1098 

We can mitigate the risk of building a "wrong" query API and not exposing the underlying messages by returning an object.  

```ts
messages.list // <- all messages as simply array (do whatever you want) 
messages.query.get() // <- query object that can be deprecated/replaced with 
messages.query2.get() // <- non breaking change query API if the API we come up with now is horrible
messages.query.experimental.pattern() // <- or hide experimental API 
```

**Pros**

- complexity of query is dealt with in core, not apps. cache invalidation could be dealt with in `setupConfig`. 

**Cons**

- moves slight complexity into core.

**Comment by felixhaeberle** [comment]
@samuelstroschein reminds me of https://www.prisma.io/docs/concepts/components/prisma-client/select-fields#select-specific-fields

**Comment by ivanhofer** [comment]
> could also be `inlang.messages.x()`, `inlang.video.x()` ?

No, we start at the `messages` node and don't have the `inlang` object. Or do we already want to think about what the API should look like in 1 year?

**Comment by samuelstroschein** [comment]
is addressed in #1149
